src/test/java/com/influxdb/v3/client/InfluxDBClientTest.java

```
@@ -21,22 +21,133 @@
  */
 package com.influxdb.v3.client;
 
+import java.io.IOException;
 import java.math.BigInteger;
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.net.URLConnection;
 import java.time.Instant;
 import java.util.Map;
 import java.util.Properties;
 import java.util.UUID;
 import java.util.stream.Stream;
+import javax.annotation.Nonnull;
 
 import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
 
+import com.influxdb.v3.client.config.ClientConfig;
 import com.influxdb.v3.client.write.WriteOptions;
 import com.influxdb.v3.client.write.WritePrecision;
 
 public class InfluxDBClientTest {
 
+    @EnabledIfEnvironmentVariable(named = "TESTING_INFLUXDB_URL", matches = ".*")
+    @EnabledIfEnvironmentVariable(named = "TESTING_INFLUXDB_TOKEN", matches = ".*")
+    @EnabledIfEnvironmentVariable(named = "TESTING_INFLUXDB_DATABASE", matches = ".*")
+    @Test
+    void testQueryWithProxy() {
+        InetSocketAddress proxyAddress = new InetSocketAddress("localhost", 10000);
+
+        try {
+            // Continue to run this test only if Envoy proxy is running in this address http://127.0.0.1:10000
+            String url = String.format("http://%s:%d/", proxyAddress.getHostName(), proxyAddress.getPort());
+            URLConnection hpCon = new URL(url).openConnection();
+            hpCon.connect();
+        } catch (IOException e) {
+            return;
```

It would be nice at least print a warning that the test is skipped due to proxy.. This way somebody could think everything is OK, but the proxy feature is not tested at all.

(Personally I don't like the `@EnabledIfEnvironmentVariable` either. We should split the tests into unit and integration tests and require these environment variables for integration tests. And not just silently skip tests.)